Questionnaire Preview UI redesign

hrit2773 opened the pull request:
## Proposed Changes

- Fixes #10679
- Fixes #12547 
- Fixes #13135 

https://github.com/user-attachments/assets/bb269e05-fcab-48f5-86cd-e1113df40e62


https://github.com/user-attachments/assets/421983a7-3b09-4ced-a092-1e7e10a6a676



@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [x] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [x] Completion of QA in Mobile Devices
- [x] Completion of QA in Desktop Devices


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
  - Added a navigation sidebar for questionnaires, allowing quick access to questions and groups, visible on larger screens.
  - Introduced a properties panel in preview mode displaying questionnaire status, subject type, version, and tags.
  - Displayed a summary of required and optional questions at the top of the preview.

- **Enhancements**
  - Improved question label styling with horizontal layout and conditional required indicators.
  - Enhanced questionnaire preview and edit layouts for better responsiveness and usability.
  - Updated navigation and group highlighting for easier question access.
  - Added smooth scrolling and toggle functionality for question navigation.
  - Improved consistency and styling of navigation buttons using UI library components.
  - Refined questionnaire form layout by conditionally rendering navigation and simplifying headers.
  - Adjusted question input element IDs for better preview mode handling.
  - Moved debug preview component inside main form container for improved layout.
  - Disabled data fetching queries in preview mode to optimize performance and prevent unnecessary requests.
  - Passed preview mode flag to historical record selectors for improved preview behavior.

- **Localization**
  - Added a new localization string for "Questionnaire Title & Description".
  - Updated "required" and "optional" labels to lowercase for consistency.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

hrit2773 committed:
questionnaire preview UI redesign

hrit2773 committed:
fixed imports

hrit2773 committed:
Merge branch 'ohcnetwork:develop' into redesign

hrit2773 commented:
@rithviknishad in order to use the Navigation in Questionniare form I made the Navigation component a reusable one. Can I get a review on it

hrit2773 commented:
I have also added required and optional question flags as mentioned in the figma design

hrit2773 committed:
Merge branch 'develop' into redesign

AdityaJ2305 commented on src/components/Questionnaire/QuestionLabel.tsx:
```diff
@@ -26,11 +26,15 @@ export function QuestionLabel({
         {(question.type === "structured" || !isSubQuestion) && (
-          <div className="hidden md:block h-1 w-4 rounded-full bg-indigo-600" />
+          <div
+            className={`hidden md:block h-1 w-4 rounded-full ${question.required ? "bg-indigo-600" : "bg-green-500"}`}
```
use cn utility here

hrit2773 replied:
made the change

AdityaJ2305 commented on src/components/Questionnaire/QuestionLabel.tsx:
```diff
@@ -26,11 +26,15 @@ export function QuestionLabel({
           {(question.type === "structured" || !isSubQuestion) && (
-            <div className="md:hidden absolute w-1 h-5 rounded-r-sm bg-indigo-500 left-3.5" />
+            <div
+              className={`md:hidden absolute w-1 h-5 rounded-r-sm ${question.required ? "bg-indigo-600" : "bg-green-500"} left-3.5`}
```
same here

hrit2773 replied:
changed

AdityaJ2305 commented on src/components/Questionnaire/QuestionRenderer.tsx:
```diff
@@ -60,6 +61,27 @@ export function QuestionRenderer({
+            </span>
+          </div>
+          <div className="flex gap-1 items-center">
+            <div className="block h-2 w-2 bg-green-500" />
```
```suggestion
            <div className="block size-2  bg-green-500" />
```

use size- instead of `h- w-` everywhere

hrit2773 replied:
changed

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
updated

hrit2773 commented:
@rithviknishad @Jacobjeevan can I get a review on this??

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

AdityaJ2305 commented on src/components/Questionnaire/QuestionnairePreviewProperties.tsx:
```diff
@@ -0,0 +1,67 @@
+                  variant="primary"
+                  className="flex items-center gap-1"
+                >
+                  <Building className="h-3 w-3" />
```
```suggestion
                  <Building className="size-3" />
```

AdityaJ2305 commented on src/components/Questionnaire/QuestionnaireNavigation.tsx:
```diff
@@ -0,0 +1,109 @@
+                      }
+                    }
+                  }}
+                  className={`w-full text-left px-3 py-2 text-sm rounded-md hover:bg-gray-200 flex items-center gap-2 ${
+                    expandedQuestions?.has(question.id) ? "bg-accent" : ""
+                  }`}
```
cn ?

AdityaJ2305 commented on src/components/Questionnaire/QuestionnaireForm.tsx:
```diff
@@ -758,14 +778,38 @@ export function QuestionnaireForm({
-                    {form.questionnaire.description}
-                  </p>
+              <div
+                className={`${encounterId === "preview" && "space-y-5 border-2 border-gray-100 rounded-lg w-full shadow-sm p-3"}`}
```
cn

hrit2773 committed:
Update src/components/Questionnaire/QuestionnairePreviewProperties.tsx

Co-authored-by: Aditya Jindal <adityajindal2305@gmail.com>

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
cn changes

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

nihal467 commented:
LGTM , will open separate issue for non-related issue

- Preview of the form before saving it itself

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

rithviknishad commented on src/components/Questionnaire/QuestionnaireNavigation.tsx:
```diff
@@ -0,0 +1,114 @@
+                {hasSubQuestions && question.questions && (
+                  <div className="ml-6 border-l-2 border-gray-200 pl-2 space-y-1">
+                    {question.questions.map((subQuestion, subIndex) => (
+                      <button
+                        key={subQuestion.id}
+                        onClick={() => {
+                          if (scrollToQuestion) {
+                            scrollToQuestion(subQuestion.id);
+                          } else if (!expandedQuestions?.has(question.id)) {
+                            if (toggleQuestionExpanded) {
+                              toggleQuestionExpanded(question.id);
```
use shadcn's button with variant prop instead? 🤔

hrit2773 replied:
@rithviknishad Ohh yaa we can do that thank's for pointing it out will update them

rithviknishad commented on src/components/Questionnaire/QuestionnaireNavigation.tsx:
```diff
@@ -0,0 +1,114 @@
+                    {
+                      "bg-accent": expandedQuestions?.has(question.id),
+                    },
+                  )}
```
here as well

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
used shadcn buttons

hrit2773 commented:
@rithviknishad addressed the changes you can go ahead with the review

hrit2773 committed:
edit form navigation made sticky

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 commented:
@rithviknishad 
https://github.com/user-attachments/assets/13e0992a-dfaa-4bbe-ba7e-ccd49090e310
resolved another related issue as well

rithviknishad commented on src/components/Questionnaire/QuestionnaireForm.tsx:
```diff
@@ -712,39 +721,50 @@ export function QuestionnaireForm({
+              />
+            ) : (
+              <div key={form.questionnaire.id} className="space-y-2">
+                <button
+                  className={cn(
+                    "w-full text-left px-2 py-1 rounded hover:bg-gray-100 font-medium",
+                    activeQuestionnaireId === form.questionnaire.id &&
+                      "bg-gray-100 text-green-600",
+                  )}
+                  onClick={() => scrollToQuestion(form.questionnaire.id)}
+                  disabled={isPending}
+                >
+                  {form.questionnaire.title}
+                </button>
+                <div className="pl-4 space-y-1">
+                  {form.questionnaire.questions
+                    .filter((q) => q.type === "group")
+                    .map((group) => (
+                      <button
+                        key={group.id}
+                        className={cn(
+                          "w-full text-left px-2 py-1 rounded text-sm hover:bg-gray-100",
+                          activeGroupId === group.id &&
+                            "bg-gray-100 text-green-600",
+                        )}
+                        onClick={() => {
+                          scrollToQuestion(form.questionnaire.id, group.id);
+                        }}
+                        disabled={isPending}
+                      >
+                        {group.text}
+                      </button>
```
i still see native html buttons!

hrit2773 replied:
@rithviknishad added outline variant buttons there as well

hrit2773 committed:
addressed buttons

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

AdityaJ2305 reviewed (changes requested):
can we use useTranslation ? instead of { t } from "i18next"

hrit2773 commented:
@AdityaJ2305 of course we can no issues with that

hrit2773 committed:
useTranslation

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 commented:
@AdityaJ2305 done

nihal467 commented:
![image](https://github.com/user-attachments/assets/af643670-91ad-4e22-a805-f145d83b2a7a)
![image](https://github.com/user-attachments/assets/c96ab974-279d-41c6-8301-ba861ff4ca08)

- the right side of the preview is completely left out
- the PR is not matching with the figma

redesign as per figma @hrit2773

hrit2773 committed:
fixed blank right

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

Jacobjeevan reviewed (changes requested):
Hmm, UI doesn't match figma design 🤔  Try and stick to the design.

Plus preview needs to work on live updates, i.e. shouldn't require user to save for preview to update.

hrit2773 commented:
@Jacobjeevan Hmm... then I have to update the edit form UI also because even that is not matching with the edit form figma

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 commented:
> @Jacobjeevan live preview will be addressed in a separate non related issue as quoted by @nihal467 earlier

or If its needed then even I can address in this PR itself... should I???

hrit2773 committed:
tried to follow figma

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 commented:
![Screenshot 2025-06-17 142807](https://github.com/user-attachments/assets/a021489c-447f-4040-8b23-0df87946f9ca)
@Jacobjeevan I have tried to follow the figma in more depth can you check it

hrit2773 commented:
added arrow corner right down arrows to the sub questions in the navigation and changed the optional and required questions color coding as per the figma and properties section is also as per the figma

Jacobjeevan commented:
> @Jacobjeevan this is the figma design ![Screenshot 2025-06-17 144158](https://private-user-images.githubusercontent.com/128292557/455911276-c3e72bc2-91ee-40ed-b708-fcfb6faddc9e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAxNTIxNDAsIm5iZiI6MTc1MDE1MTg0MCwicGF0aCI6Ii8xMjgyOTI1NTcvNDU1OTExMjc2LWMzZTcyYmMyLTkxZWUtNDBlZC1iNzA4LWZjZmI2ZmFkZGM5ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYxN1QwOTE3MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNDQ4NWY2ODA4ODdlYTc3M2VkODdiY2M1YmFiNmE1ZmNjZGI2OTMwNDc1MjIwZDU4ZjdmNDI0ZWExYmI2NDJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.TG74orik2IEt7M_Vybkr_KUTyuG6RkpViBHX2fpYXC4)

<img width="1010" alt="image" src="https://github.com/user-attachments/assets/89353386-1b7e-4a48-af17-92e5bcef0370" />

You don't see any differences between figma and the current UI? 😅  Please go through everything before requesting review/pinging again.

Also check for mobile as well.

hrit2773 committed:
navigation fixed back

hrit2773 committed:
Merge branch 'redesign' of https://github.com/hrit2773/care_fe into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
changed to figma

hrit2773 committed:
Merge branch 'redesign' of https://github.com/hrit2773/care_fe into redesign

AdityaJ2305 commented:
@hrit2773 , Great with design work 

few things :
- [ ] Navigation is not working for few questions in preview (tested on : Community Homecare form)
- [ ] Navigation designed needs to be consistent for encounter form filling

hrit2773 commented:
@AdityaJ2305 should I use the same navigation component for encounter form also??

hrit2773 committed:
added special classes for preview questions

hrit2773 committed:
fixed accidental

hrit2773 committed:
mistake

hrit2773 committed:
removed native buttons

hrit2773 commented:
@rithviknishad @Jacobjeevan can I get a review on the changes

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
resolved conflicts-1

hrit2773 commented:
@Jacobjeevan @rithviknishad I have resolved conflicts also and made changes. Can I get a review on it

hrit2773 commented:
@AdityaJ2305 Alright

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
used translation

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
merge conflicts resolved

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 commented:
@rithviknishad @Jacobjeevan final review on this for merging?

hrit2773 committed:
Merge branch 'develop' into redesign

Jacobjeevan commented on src/components/Questionnaire/QuestionnaireNavigation.tsx:
```diff
@@ -0,0 +1,96 @@
+                  variant={null}
+                  onClick={() => {
+                    scrollToQuestion(
+                      isPreview ? question.id : question.link_id,
```
why id here for preview? 🤔

Jacobjeevan commented on src/components/Questionnaire/QuestionnaireNavigation.tsx:
```diff
@@ -0,0 +1,96 @@
+                    {question.questions.map((subQuestion, subIndex) => (
+                      <Button
+                        key={subQuestion.link_id}
+                        variant={null}
```
🤔  Let's not, use one of the variants, ghost/white might work.

Jacobjeevan commented on src/components/Questionnaire/QuestionnaireNavigation.tsx:
```diff
@@ -0,0 +1,96 @@
+                            }, 100);
+                          } else {
+                            scrollToQuestion(
+                              isPreview ? subQuestion.id : subQuestion.link_id,
```
let's stick with using link id

Jacobjeevan commented on src/components/Questionnaire/QuestionnaireForm.tsx:
```diff
@@ -746,64 +748,58 @@ export function QuestionnaireForm({
+                  .filter((q) => q.type === "group")
+                  .map((group) => (
+                    <Button
+                      variant={null}
```
same here.

Jacobjeevan commented on src/components/Questionnaire/QuestionnaireForm.tsx:
```diff
@@ -746,64 +748,58 @@ export function QuestionnaireForm({
+          {questionnaireForms.map((form) => (
+            <div key={form.questionnaire.id} className="space-y-2">
+              <Button
+                variant={null}
```
white/ghost or one of the other variants

Jacobjeevan commented on src/components/Questionnaire/QuestionnaireEditor.tsx:
```diff
@@ -856,6 +858,17 @@ export default function QuestionnaireEditor({ id }: QuestionnaireEditorProps) {
 
+  const scrollToPreviewQuestion = (groupId?: string) => {
+    setPreviewGroupId(groupId);
+    const element = document.querySelector(
```
since the tabs are exclusive, you should be able to reuse scrollToQuestion function from utils

Jacobjeevan commented on src/components/Questionnaire/QuestionTypes/QuestionGroup.tsx:
```diff
@@ -159,6 +161,7 @@ export const QuestionGroup = memo(function QuestionGroup({
         </div>
       )}
       <div
+        data-preview-question-id={encounterId === "preview" && question.id}
```
shouldn't be necessary once you switch to using scrollToQuestion fn

Jacobjeevan commented on src/components/Questionnaire/QuestionRenderer.tsx:
```diff
@@ -60,8 +61,32 @@ export function QuestionRenderer({
+      )}
       {questions.map((question) => (
         <div
+          data-question-id={question.id}
```
not needed, stick to using link Id for scroling.

Jacobjeevan reviewed (changes requested):
Check preview styling against figma - minor adjustments are needed. Also check mobile as well.

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
minor changes in styling against figma

hrit2773 committed:
made requested changes

hrit2773 committed:
button variant used

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 commented:
@Jacobjeevan made the changes as requested also fixed some styling issues to match the figma. Fixed for mobile as well seems to be working fine. And also used link_id instead of id and questionnaire Navigation component is same as that of the original one and also changed the button variant to ghost. Reused the utils scroll to question

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

nihal467 commented:
<img width="1918" height="1018" alt="image" src="https://github.com/user-attachments/assets/41336972-13f9-43d5-b436-3ac0d8a8148a" />

[master-questionnaire (3).json](https://github.com/user-attachments/files/21426223/master-questionnaire.3.json)

When we're opening the question, multiple question's are triggering request, can you recheck it and mae sure there is no unwanted API are fired

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
removed unwanted api calls fromm questions

hrit2773 committed:
removed unwanted api calls fromm questions1

hrit2773 commented:
@nihal467 can you check it now I fixed those 404 not found triggered due to some unwanted calls

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 commented:
@nihal467 referenced this PR to a new issue #13135

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 commented:
@nihal467 can you test it again.

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 commented:
@rithviknishad @Jacobjeevan can I get review on this

hrit2773 committed:
Merge branch 'develop' into redesign

nihal467 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
final changes with some unwanted api requests removal

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 commented:
@nihal467 made changes to navigation text color and questionnaire title description font to align and adjusted the spacing as well as disabled the unwanted api requests from historical records component in preview

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
force push

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 committed:
Merge branch 'develop' into redesign

hrit2773 commented:
@nihal467 can you test this again

hrit2773 commented:
@rithviknishad @Jacobjeevan can you review this

nihal467 commented:
<img width="1299" height="362" alt="image" src="https://github.com/user-attachments/assets/e5ca8ecf-98d9-40b9-ac59-a6984012a0b7" />

- can you compare this portion with the figma design, do make sure you at least do a self-review before requesting testing

Jacobjeevan commented:
Closing PR; we are restricting questionnaire changes to core team.
